docs/getting-started/auth-self-defence-checks.adoc

```diff
@@ -0,0 +1,188 @@
+
+==== Synchronously
+
+To run multiple self defence checks _synchronously_, you'll first need to invoke the link:javadoc[`SecurityService#getCheckExecutor`] method.  This methods returns link:javadoc[`SyncSecurityCheckExecutor`] where checks can be added (chained) to the executor and executed synchronously.
+
+
+[source, java]
+----
+SyncSecurityCheckExecutor syncCheckExecutor = securityService.getCheckExecutor();
+----
+
+==== Asynchronously
+
+To run multiple self defence checks _asynchronously_, you’ll first need to invoke the link:javadoc[`SecurityService#getAsyncCheckExecutor`] method. This methods returns link:javadoc[`AsyncSecurityCheckExecutor`] where checks can be added (chained) to the executor and executed asynchronously.
+
+[source, java]
+----
+AsyncSecurityCheckExecutor asyncCheckExecutor = securityService.getAsyncCheckExecutor();
+----
+
+==== Adding Checks to Self Defence Check Executors
+
+Both the link:javadoc[`SyncSecurityCheckExecutor`] and link:javadoc[`AsyncSecurityCheckExecutor`] have an `addCheck` method.  This method expects either link:javadoc[`SecurityCheckType`]  or link:javadoc[`SecurityCheck`] as a paramater. +
+
+If you are invoking the `addCheck` method on a link:javadoc[`SyncSecurityCheckExecutor`] then it will return link:javadoc[`SyncSecurityCheckExecutor`]. +
+
+If you are invoking the `addCheck` method on an link:javadoc[`AsyncSecurityCheckExecutor`] then it will return link:javadoc[`AsyncSecurityCheckExecutor`].
+
+[source, java]
+----
+// adding self defence checks to a SyncSecurityCheckExecutor
+
+// add two provided self defence checks to the syncCheckExecutor
+syncCheckExecutor.addCheck(SecurityCheckType.<check_type>).addCheck(SecurityCheckType.<check_type>);
+// add the CustomSecurityCheck to the syncCheckExecutor
+syncCheckExecutor.addCheck(new customSecurityCheck());
+
+
+// adding self defence checks to a AsyncSecurityCheckExecutor
+
+// add two provided self defence checks to the asyncCheckExecutor
+asyncCheckExecutor.addCheck(SecurityCheckType.<check_type>).addCheck(SecurityCheckType.<check_type>);
+// add the CustomSecurityCheck to the asyncCheckExecutor
+asyncCheckExecutor.addCheck(new customSecurityCheck());
+----
+
+==== Executing Self Defence Checks on Executors
+
+Both link:javadoc[`SyncSecurityCheckExecutor`] and link:javadoc[`AsyncSecurityCheckExecutor`] have an `execute` method that executes all self defence checks that have been added. +
+
+The `execute` method for link:javadoc[`SyncSecurityCheckExecutor`] returns a `Map` where the key is the name of the self defence check being tested (`String`) and the value is  link:javadoc[`SecurityCheckResult`]. +
+
+The `execute` method for link:javadoc[`AsyncSecurityCheckExecutor`] returns a `Map` where the key is the name of the self defence check being tested (`String`) and the value is a `Map` of type `Future` with link:javadoc[`SecurityCheckResult`].
```

Comment:
Maybe worths saying that the name of the check is the result of the `getName()` method on the enum or on the check instance.